packages/fuels-programs/src/call_utils.rs

```diff
@@ -38,12 +40,23 @@ pub async fn build_tx_from_contract_calls(
     wallet: &WalletUnlocked,
 ) -> Result<ScriptTransaction> {
     let consensus_parameters = wallet.get_provider()?.consensus_parameters().await?;
+    let n_vectors_calls = calls.iter().filter(|c| c.output_param.is_vector()).count();
+
+    // Compute the length of the calling scripts for the two types of contract calls.
+    // Use placeholder for `call_param_offsets` and `output_param_type`, because the length of the
+    // calling script doesn't depend on the underlying type, just on whether or not the contract
+    // call output is a vector.
+    let calls_instructions_len_no_vectors =
+        get_single_call_instructions(&CallOpcodeParamsOffset::default(), &ParamType::U64).len()
+            * (calls.len() - n_vectors_calls);
+    let calls_instructions_len_vectors = get_single_call_instructions(
+        &CallOpcodeParamsOffset::default(),
+        &ParamType::Vector(Box::from(ParamType::U64)),
+    )
+    .len()
+        * n_vectors_calls;
 
-    // Calculate instructions length for call instructions
-    // Use placeholder for call param offsets, we only care about the length
-    let calls_instructions_len =
-        get_single_call_instructions(&CallOpcodeParamsOffset::default()).len() * calls.len();
-
+    let calls_instructions_len = calls_instructions_len_no_vectors + calls_instructions_len_vectors;
```

Comment:
Maybe the logic for calculating this can be moved to a helper. I feel like it's a bit neater if the comment becomes a docstring for the helper.